src/main/java/ro/msg/learning/shop/service/ProductService.java

```diff
@@ -0,0 +1,82 @@
+package ro.msg.learning.shop.service;
+
+import jakarta.persistence.EntityNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.domain.Product;
+import ro.msg.learning.shop.domain.ProductCategory;
+import ro.msg.learning.shop.dto.ProductDto;
+import ro.msg.learning.shop.repository.ProductRepository;
+
+import java.util.List;
+import java.util.UUID;
+
+@RequiredArgsConstructor
+@Service
+public class ProductService {
+    public static final String NO_PRODUCT_IS_FOUND = "No product was found!";
+
+    private final ProductCategoryService productCategoryService;
+    private final ProductRepository productRepository;
+
+
+    public Product getProductById(UUID productId) {
+        Product product = productRepository.findById(productId).orElse(null);
+        if (product != null) {
+            return product;
+        }
+        throw new EntityNotFoundException(NO_PRODUCT_IS_FOUND);
+    }
+
+    public List<Product> getAllProducts() {
+        List<Product> allProducts = productRepository.findAll();
+        if (!allProducts.isEmpty()) {
+            return allProducts;
+        }
+        throw new EntityNotFoundException(NO_PRODUCT_IS_FOUND);
+    }
+
+    public Product createProduct(ProductDto productToCreate) {
```

Comment:
If you return an entity from service to controller, you should also receive an entity from the controller to the service. 

If you send a DTO from the controller to the service, you should also return a DTO from the service to the controller.

Now you code is mixed. Both service and controller use both entities and DTOs, which is not right.